### DISCUSSION TITLE
Steadily removing D-Bus

### DISCUSSION THREAD

**Discussion by veracioux** [flameshot-org/flameshot#1999] [General]
@borgmanJeremy @mmahmoudian I have started working on removing D-Bus (where possible/desirable) and I'll submit a PR fairly soon. The current results look promising.

I'm going to use this discussion to ask some questions, so I don't break something in the process.

**Comment by veracioux** [comment]
@borgmanJeremy @mmahmoudian What are all the uses for `FlameshotDBusAdapter::captureSaved`, `captureTaken` and `captureFailed` externally (outside of the source code, which I can easily inspect)?

I know that `capture{Taken,Failed}` are used to print the screenshot or geometry when the `--print-geometry` or `--raw` options are used. But other than that, is there another use for them? The `captureSaved` seems to be unused.

How I checked:
- grep
- using `dbus-monitor`, but I'm still a bit clumsy with it

**Reply by borgmanJeremy** [reply]
No I dont believe they are used anywhere externally.

**Comment by ElTh0r0** [comment]
As mentioned in PR #4211, the KDSingleApplication message feature _could_ replace `flameshotdbusadapter` for all platforms. So I am raising the question if we should consider to remove `flameshotdbusadapter` completely and use KDSingleApplication messages on all platforms? For sure D-Bus will remain in general for the Linux builds, as D-Bus is used e.g. by `ScreenGrabber`. 

The advantage would be that only one solution would need to be maintained across all platforms for the CLI commands from a second instance. Disadvantage: KDSingleApplication would be a 'hard' exernal dependency (meaning it would always be necessary to be included).

Unknown (to me): Would this change affect Flatpak/snaps/AppImage? I really don't know, if they are currently using / are able use the D-Bus connection to forward CLI commands from a second instance; neither if KDSingleApplication message will work for them.

**Reply by borgmanJeremy** [reply]
I think flatpak at least should still work with dbus is substituted with with KDSingleApplication because the flatpak properly opens only a single instance. As you say DBus must remain to interface with the portal on wayland, but it would be really nice to consolidate the other dbus stuff to be the same across all platforms.
